### PR TITLE
Action menu

### PR DESCRIPTION
**Pull request by wpwood**
this PR address https://github.com/rackerlabs/canon-react/issues/15

### CODE REVIEW

**Comment by wpwood** [comment]
@willcumbie @jdrush89 @adampickeral 

I resurrected an old branch that Josh was working on and brought it up to date with the latest master. Thoughts?

**Comment by adampickeral** [comment]
we have es6 support now. change it to use classes instead of `React.createClass`

File: src/DropdownItem.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+var React = require('react');
+
+var DropdownItem = React.createClass({
+  propTypes: {
+    enabled: React.PropTypes.bool,
+    onClick: React.PropTypes.func,
+    type: React.PropTypes.string
```
Comment (adampickeral) [inline src/DropdownItem.jsx]: seems like there are a limited number of types, so this should be `React.PropTypes.oneOf(['link', 'category', 'text'])`

File: src/DropdownItem.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+var React = require('react');
+
+var DropdownItem = React.createClass({
+  propTypes: {
+    enabled: React.PropTypes.bool,
+    onClick: React.PropTypes.func,
```
Comment (adampickeral) [inline src/DropdownItem.jsx]: .isRequired

File: src/Dropdown.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      classes.push(dropdownTypes[this.props.type]);
+    }
+
+    return classes.join(' ');
+  },
+
+  render: function () {
+    var style;
+
+    style = {float: 'left'};
```
Comment (adampickeral) [inline src/Dropdown.jsx]: why is this needed?

File: src/DropdownItem.jsx (hunk 2 above)
Comment (adampickeral) [inline src/DropdownItem.jsx]: actually, i wonder if this should be optional, and only applied if the type is `link`

File: src/DropdownItem.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    itemClasses = ['rs-dropdown-item'];
+    if (this.props.className) {
+      itemClasses.push(this.props.className);
+    }
+    if (!this.props.enabled) {
+      itemClasses.push('disabled');
+    }
+
+    return (
+      <li {...this.props} className={itemClasses.join(' ')} onClick={this._handleClick}>
```
Comment (adampickeral) [inline src/DropdownItem.jsx]: `onClick` should only be added for links, right?

File: src/DropdownItem.jsx (hunk 2 above)
Comment (wpwood) [inline src/DropdownItem.jsx]: It was optional, and not supplied on the demo, resulting in an 'undefined' error when clicked. That's why I gave it a default value. I think it should be optional, and if not supplied, just a no-op.

**Comment by wpwood** [comment]
@adampickeral I looked at only passing onClick for links. The problem is that (a) we pass props as a spread attribute so that other HTML attributes can be used, and (b) onClick is valid for other DOM elements. We would have to filter it out of props explicitly, and possibly break it for other uses. Filtering is not straightforward, either. There's an experimental ES7 syntax that JSX supports, but it requires a JSX flag.

Anyway, I think that it's more trouble than it's worth.

**Comment by wpwood** [comment]
@willcumbie @jdrush89 @adampickeral Any other comments?
